closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityProperties.java

```diff
@@ -0,0 +1,707 @@
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2022, 2022 All Rights Reserved
+ * ===========================================================================
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * IBM designates this particular file as subject to the "Classpath" exception
+ * as provided by IBM in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * ===========================================================================
+ */
+
+package openj9.internal.security;
+
+import java.security.Provider.Service;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Properties;
+
+import sun.security.util.Debug;
+
+public final class RestrictedSecurityProperties {
+
+    private static final Debug debug = Debug.getInstance("semerufips");
+
+    private static RestrictedSecurityProperties instance;
+
+    private static String descName;
+    private static String descNumber;
+    private static String descPolicy;
+    private static String descSunsetDate;
+
+    // Security properties.
+    private static String jdkTlsDisabledNamedCurves;
+    private static String jdkTlsDisabledAlgorithms;
+    private static String jdkTlsDphemeralDHKeySize;
+    private static String jdkTlsLegacyAlgorithms;
+    private static String jdkCertpathDisabledAlgorithms;
+    private static String jdkSecurityLegacyAlgorithm;
+    private static String keyStoreType;
+    private static String keyStore;
+
+    // For Secure Random.
+    private static String jdkSecureRandomProvider;
+    private static String jdkSecureRandomAlgorithm;
+
+    // Provider with argument (provider name + optional argument).
+    private static ArrayList<String> providers = new ArrayList<String>();;
+    // Provider without argument.
+    private static ArrayList<String> providersSN = new ArrayList<String>();;
+    // Constraints for each provider. Key is the Provider Name, Value is the Constraints.
+    private static Map<String, String[][]> providerConstraints = new HashMap<String, String[][]>();
```

Comment:
Please define fields using interfaces (`List`, not `ArrayList`), avoid redundant type parameters and superfluous semicolons:
```java
    private static List<String> providers;
    // Provider without argument.
    private static List<String> providersSN;
    // Constraints for each provider if there are.
    private static Map<String, Constraint[]> providerConstraints;
```
Initialization should occur in the constructor.
Note also the use of a nested class:
```java
    private static final class Constraint {
        final String type;
        final String algorithm;
        final String attributes;

        Constraint(String type, String algorithm, String attributes) {
            super();
            this.type = type;
            this.algorithm = algorithm;
            this.attributes = attributes;
        }

    }
```